webpack/assets/javascripts/foreman_users.test.js

```diff
@@ -70,3 +70,40 @@ describe('taxonomyAdded', () => {
     expect($('#user_default_organization_id option').length).toEqual(1);
   });
 });
+
+describe('users Password', () => {
+  it('should toggle password field', () => {
+    document.body.innerHTML = `
+    <div>
+      <select onchange="users.authSourceSelected(this)" name="user[auth_source_id]"
+        id="user_auth_source_id">
+        <option value=""></option>
+        <option value="8">LDAP-QA</option>
+        <option value="9">INTERNAL</option>
+        <option value="10">LDAP-CORP</option>
+      </select>
+      <div id="password" style="">
+        <div id="password_react">
+        <div>
+          <div class="form-group  ">
+          <label class="col-md-2 control-label">Password</label>
+            <div class="col-md-4">
+              <div class="ReactPasswordStrength  ">
+                <input type="password" name="user[password]" id="user_password" class="ReactPasswordStrength-input form-control" value="">
+                  <div class="ReactPasswordStrength-strength-bar">
+               </div><span class="ReactPasswordStrength-strength-desc">Too short</span></div>
+            </div>
+          </div>
+          <div class="form-group  "><label class="col-md-2 control-label">Verify</label>
+            <div class="col-md-4"><input id="password_confirmation" name="user[password_confirmation]" type="password" class="form-control"></div>
+          </div>
+        </div>
+      </div>
+    </div>`;
+
+    $('#user_auth_source_id')
+      .val('9')
+      .change();
+    expect($('#password.input').length).toEqual(2);
```

Comment:
`.length` does not care whether the element is visible or not, and in this case will also never be 2 (since you are selecting an element by id). The selector you are looking for is `#password:visible`, which should return the div when it's visible and empty list when it isn't. Also, you set the div to visible initially so it doesn't check that the function works for showing - perhaps check both ways.